src/components/Person.js

```diff
@@ -5,150 +5,136 @@ import store from 'store2';
 import { updated_niceties_spinlock } from "./People";
 
 
-const Person = React.createClass({
-    getInitialState: function() {
-        let textValue = '';
-        let anonValue = false;
-        let noReadValue = false;
-        let dataPerson;
-        let foundPerson = false;
-        for (var i = 0; i < this.props.fromMe.length; i++) {
-            if (this.props.fromMe[i].target_id === this.props.data.id) {
-                dataPerson = this.props.fromMe[i];
-                foundPerson = true;
-                break;
-            }
-        }
-        let dateUpdated;
-        if (foundPerson) {
-            if (dataPerson.date_updated === null) {
-                dateUpdated = '';
-            } else {
-                dateUpdated = dataPerson.date_updated;
-            }
+class Person extends React.Component {
+
+  constructor(props) {
```

Comment:
This method is kind of a mess. I think you've handled it correctly, but trying to understand it, it appears that one of its props is a complete list of niceties the current user has written? That shouldn't be the data it gets, and it shouldn't have to try to pick out the maybe-present single relevant nicety.

I'd certainly buy the argument that changing that is out of scope for this change, but if you want to solve it this feels like a good time to do so.

Reply:
Do you want to move the nicety-selecting logic into PeopleRow, the parent component? I think it is out of scope but I can start thinking about how I might do that.

Reply:
I suppose that is what I'm suggesting! And yeah, we can and should handle that separately; good call.